### PR TITLE
Add scheduler option to `assert_eq` utilities

### LINKED ISSUE #8583: Add scheduler option to `assert_eq` utilities
In https://github.com/dask/dask/pull/8559 we added a `scheduler=` keyword argument to `dask.dataframe.assert_eq` to allow for cases where we want to run the computes that happen in `assert_eq` on a specified scheduler. As a follow-up, we should add a similar `scheduler=` keyword argument to our other `assert_eq` utilities in `dask.array`

https://github.com/dask/dask/blob/a504115312b1e591604276a3098b2e0076bf49e1/dask/array/utils.py#L261

and `dask.bag`

https://github.com/dask/dask/blob/a504115312b1e591604276a3098b2e0076bf49e1/dask/bag/utils.py#L1

### PR DESCRIPTION
**Pull request by xinrong-databricks**
- [x] Closes #8583
- [x] Tests added / passed
- [x] Passes `pre-commit run --all-files`

### CODE REVIEW

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

**Comment by xinrong-databricks** [comment]
Are test failures above infra-related?

**Comment by jsignell** [comment]
Yeah the test failures mostly have to do with dask not being quite caught up with the latest release of pandas: #8580

**Comment by jsignell** [comment]
ok to test

**Comment by xinrong-databricks** [comment]
Thanks @jsignell !

File: dask/array/tests/test_array_utils.py [hunk 1]
```diff
@@ -87,3 +89,29 @@ def test_meta_from_array_type_inputs():
...
+        return get_sync(*args, **kwargs)
+
+    assert_eq(a, b)
+    assert counter == 0
+
+    assert_eq(a, b, scheduler=custom_scheduler)
+    if isinstance(a, Array) and isinstance(b, Array):
+        assert counter == 2
+    else:
+        assert counter == 0
```
Comment (jsignell) [inline dask/array/tests/test_array_utils.py:117]: I really like this test! But this part took me a minute to understand. I am wondering if something like this is clearer:

```suggestion
    n_da_arrays = len([x for x in [a, b] if isinstance(x, Array)])
    assert counter == n_da_arrays
```

**Review by jsignell — COMMENTED** [review]
This looks great! I just have one small suggestion

File: dask/array/tests/test_array_utils.py (hunk 1 above)
Comment (xinrong-databricks) [reply dask/array/tests/test_array_utils.py]: That's so smart! Committed, thank you.

**Comment by jsignell** [comment]
Thanks @xinrong-databricks! This is in :)

**Comment by xinrong-databricks** [comment]
Thank you @jsignell !

**Pull request merged**